cmd/opentelemetry-collector/app/exporter/cassandra/factory.go

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2020 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cassandra
+
+import (
+	"fmt"
+
+	"github.com/open-telemetry/opentelemetry-collector/config/configerror"
+	"github.com/open-telemetry/opentelemetry-collector/config/configmodels"
+	"github.com/open-telemetry/opentelemetry-collector/exporter"
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/plugin/storage/cassandra"
+)
+
+const (
+	// TypeStr defines type of the Cassandra exporter.
+	TypeStr = "jaeger_cassandra"
+)
+
+// OptionsFactory returns initialized cassandra.OptionsFactory structure.
+type OptionsFactory func() *cassandra.Options
+
+// DefaultOptions creates Cassandra options supported by this exporter.
+func DefaultOptions() *cassandra.Options {
+	return cassandra.NewOptions("cassandra")
+}
+
+// Factory is the factory for Jaeger Cassandra exporter.
+type Factory struct {
+	OptionsFactory OptionsFactory
+}
+
+// Type gets the type of exporter.
+func (Factory) Type() string {
+	return TypeStr
+}
+
+// CreateDefaultConfig returns default configuration of Factory.
+func (f Factory) CreateDefaultConfig() configmodels.Exporter {
```

Comment:
Is the name CreateDefaultConfig required by some external API?  We usually mention it in the comment.

Given that the result of this function depends on externally provided (via OptionsFactory) options, the name "default" is confusing.

Reply:
`CreateDefaultConfig()` is part of the `extension.Factory` interface in the Otel collector. I agree that "default" is confusing here, we should ideally apply user supplied options once the default config is created

Reply:
>  I agree that "default" is confusing here, we should ideally apply user supplied options once the default config is created

OTEL service applies config file changes to the returned instance. We cannot override them by config supplied as legacy jaeger flags. The OTEL config takes precedence over legacy jaeger configuration.